apps/search/js/providers/suggestions.js

```diff
@@ -37,13 +37,15 @@
     },
 
     search: function(input) {
-      return new Promise((resolve, reject) => {
-        var url = encodeTerms(SearchProvider('suggestUrl'), input);
-        LazyLoader.getJSON(url, true).then(result => {
-          var results = result[1];
-          // We add an item to search the entered term as well
-          results.unshift(result[0]);
-          resolve(results);
+      return SearchProvider.ready().then(function() {
+        return new Promise((resolve, reject) => {
```

Comment:
err no you cant do that, .then resolves with the value of the promise its called on